ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -352,9 +352,17 @@ export const ApplicationResourceTree = (props: ApplicationResourceTreeProps) =>
                 childrenByParentKey.set(treeNodeKey(parent), children);
             });
         });
-        roots = nodes.filter(node => (node.parentRefs || []).length === 0).sort(compareNodes);
+        roots = nodes.filter(node => (node.parentRefs || []).length === 0 || isManagedResource(node)).sort(compareNodes);
     }
 
+    function isManagedResource(node: ResourceTreeNode) {
```

Comment:
Minor improvement. Please store keys of `props.app.status.resources` in Set to avoid `n*m` complexity.